File: Drivo.Entities/StudentEntity.cs
namespace Drivo.Entities
{
    public class StudentEntity : UserEntity
    {
        public StudentEntity() : base()
        {

        }

        public StudentEntity(string userName, string email, string firstName, string lastName, DateTime birthDate, string phoneNumber) : base(userName, email, firstName, lastName, birthDate, phoneNumber)
        {

        }

        public virtual StudentsGroupEntity? StudentsGroup { get; set; }
        public int? StudentsGroupId { get; set; }
        public virtual InstructorEntity? Instructor { get; set; }
        public int? InstructorId { get; set; }
        public virtual List<DrivingEntity>? Drivings { get; set; }
        public virtual List<InternalExamEntity>? InternalExams { get; set; }
        public virtual List<ExternalExamEntity>? ExternalExams { get; set; }
        public virtual List<PaymentEntity>? Payments { get; set; }
    }
}

File: Drivo/src/entities/StudentEntity.ts
import { DrivingEntity } from "./DrivingEntity";
import { ExternalExamEntity } from "./ExternalExamEntity";
import { InstructorEntity } from "./InstructorEntity";
import { InternalExamEntity } from "./InternalExamEntity";
import { PaymentEntity } from "./PaymentEntity"
import { StudentsGroupEntity } from "./StudentsGroupEntity";
import { UserEntity } from "./user.entity"

export class StudentEntity extends UserEntity {
    studentsGroup: StudentsGroupEntity | null;
    studentsGroupId: number | null;
    instructor: InstructorEntity | null;
    instructorId: number | null;
    drivings: Array<DrivingEntity> | null;
    internalExams: Array<InternalExamEntity> | null;
    externalExams: Array<ExternalExamEntity> | null;
    payments: Array<PaymentEntity> | null;
}